renderer/renderer.go

```diff
@@ -164,11 +170,39 @@ func AtomicWrite(path string, createDestDirs bool, contents []byte, perms os.Fil
 
 	// If we got this far, it means we are about to save the file. Copy the
 	// current contents of the file onto disk (if it exists) so we have a backup.
-	if backup {
-		if _, err := os.Stat(path); !os.IsNotExist(err) {
-			if err := copyFile(path, path+".bak"); err != nil {
+	if backup && maxBackup > 0 {
+		if maxBackup == 1 {
+			if _, err := os.Stat(path); !os.IsNotExist(err) {
+				if err := copyFile(path, path+".bak"); err != nil {
+					return err
+				}
+			}
+		} else {
+			if _, err := os.Stat(path); !os.IsNotExist(err) {
+				if err := copyFile(path, path+"."+strconv.FormatInt(time.Now().UnixNano()/1000, 10)); err != nil {
```

Comment:
I think if you're going to add a timestamp, you might as well make it human readable. So instead of UnixNano truncated to microseconds, maybe something like...

path+"."+time.Now().Format("20060102150405.0000+".bak")

I cut it off at hundreds of micro-seconds, as that seems enough, and I added the '.bak' as it seems like all back files should have the same suffix.

Reply:
ok